`find/2` input protection proposal.

Hajto opened the pull request:
Today I had an issue where I accidentally passed `{:ok, tree}` into the `find/2`. I think it might not be a bad idea to protect the input with a guard. People who can't read the docs like me might save some frustration.

This is purely proposal. If I get a green light for this I will make develop this PR properly.

philss commented on lib/floki.ex:
```diff
@@ -1,4 +1,5 @@
 defmodule Floki do
+  require Floki
```
This is not needed :)
```suggestion
```

Hajto replied:
Sorry, some sort of brain fart.

Hajto replied:
VS Code added that line accidentally. I removed it. Good catch.

philss reviewed (commented):
I think it's a good idea. Please go ahead and try to fix the tests :)

Hajto commented:
<img width="365" alt="image" src="https://github.com/philss/floki/assets/8336429/a8139b4f-e5e4-4819-92b0-108e1f6bbbb0">
Should this be deleted?

Hajto commented:
Added: 
 - Tests
Fixed:
 -  wrong doctype specification (it looks like it will never affect the `find/2` guard but fixed for completenes)
 -  moved list match out of guard definition since list is not valid node

philss commented:
@Hajto thank you! :purple_heart:
